### PR TITLE
Feat: msat to_approx_str

### PR DESCRIPTION
**Pull request by m-schmoock**
I reduced PR #2630 to just contain a reduced version of `to_approx_str`  and its testcase.
So you can quickly make a decision :D

### CODE REVIEW

File: contrib/pylightning/tests/test_units.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from lightning import Millisatoshi
+import pytest
```
Comment (rustyrussell) [inline contrib/pylightning/tests/test_units.py]: flake8 is upset:

    contrib/pylightning/tests/test_units.py:2:1: F401 'pytest' imported but unused

Comment (m-schmoock) [reply contrib/pylightning/tests/test_units.py]: Whoops. Fixed

**Comment by rustyrussell** [comment]
This surprised me, that it lost precision.  I think this needs to be called "to_approx_str" maybe?

**Comment by m-schmoock** [comment]
Good point, renamed it

**Comment by m-schmoock** [comment]
Note: I added the PR that adds this and `mbtc` unit in https://github.com/ElementsProject/lightning/pull/2657

I leave this open for discussion.

**Comment by rustyrussell** [comment]
I like it, but I'm reluctant to apply with @cdecker since he's the pylightning owner....

**Comment by m-schmoock** [comment]
No changes, I just rebased to master, just so #2657 can build on this, as it had merge conflicts.

**Comment by m-schmoock** [comment]
ACK 7edfb60  (since no change, just rebase since rusty's ack)

**Comment by m-schmoock** [comment]
ACK 7edfb60

File: contrib/pylightning/lightning/lightning.py [hunk 2]
```diff
@@ -85,6 +86,23 @@ def to_btc_str(self):
         else:
             return '{:.8f}btc'.format(self.to_btc())
 
+    def to_approx_str(self, digits: int = 3):
+        """
```
Comment (cdecker) [inline contrib/pylightning/lightning/lightning.py]: Comments usually have the format

```python
"""Short one-line description

Longer, potentially multi-line description
"""
```

File: contrib/pylightning/tests/test_units.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    amount = Millisatoshi('1000sat')
+    assert amount.to_approx_str() == "1000sat"
+    amount = Millisatoshi('100msat')
+    assert amount.to_approx_str() == "0.1sat"
+
+    # also test significant rounding
+    amount = Millisatoshi('10001234sat')
+    assert amount.to_approx_str() == "0.1btc"
+    amount = Millisatoshi('1234sat')
+    assert amount.to_approx_str() == "1230sat"
```
Comment (cdecker) [inline contrib/pylightning/tests/test_units.py]: This is kind of weird, we don't gain a length advantage by rounding the last digit, but we do lose precision.

Comment (cdecker) [reply contrib/pylightning/tests/test_units.py]: I think the `digit` parameter should allow rounding, but if it doesn't give an advantage length-wise it should force it.

File: contrib/pylightning/lightning/lightning.py (hunk 2 above)
Comment (m-schmoock) [reply contrib/pylightning/lightning/lightning.py]: thx for pointing out

File: contrib/pylightning/tests/test_units.py (hunk 3 above)
Comment (m-schmoock) [reply contrib/pylightning/tests/test_units.py]: fixed

**Comment by cdecker** [comment]
ACK 3c35f79ec16bc28bf61bd6f919d64d5a656087f7

**Pull request merged**